src/storage/storage.cpp

```diff
@@ -165,6 +170,81 @@ aos::RetWithError<aos::sm::servicemanager::ServiceData> Storage::GetService(cons
     return aos::RetWithError<aos::sm::servicemanager::ServiceData>(ConvertServiceData(serviceData));
 }
 
+aos::Error Storage::AddCertInfo(const aos::String& certType, const aos::iam::certhandler::CertInfo& certInfo)
+{
+    aos::LockGuard lock(mMutex);
+
+    LOG_DBG() << "Add cert info: " << certType;
+
+    return mCertDatabase.Add(ConvertCertInfo(certType, certInfo));
```

Comment:
It looks like task added by  "[storage] Return error if record exist" commit, isn't properly implemented: we should return error if key field/fields already exists. Key field/fields are different in each DB. We need to reimplement it. I've reopened the task.

Reply:
Ok